### DISCUSSION TITLE
Android : Gradle Plugin versions and Java version error

### DISCUSSION THREAD

**Discussion by pol2095** [airsdk/Adobe-Runtime-Support#3285] [Q&A] [answered]
Hello,

using AIR SDK 51.0.1.2, `adt` return this error on Android :
```
NOTE: Multiple Android build-tools folders exist, choosing the latest version (34.0.0) for packaging
NOTE: targets later than Android 33 need Gradle Plugin versions of 8 or more. Adjusting to 8.1.1
unexpected failure: Unable to run java: com.adobe.air.ADTException: gradle tool failed:
FAILURE: Build failed with an exception.

* Where:
Build file 'C:\Users\user\AppData\Local\Temp\136975d6-c005-45d1-8b89-c13d3a0aa1a1\app\build.gradle' line: 1

* What went wrong:
An exception occurred applying plugin request [id: 'com.android.application']
> Failed to apply plugin 'com.android.internal.application'.
   > Android Gradle plugin requires Java 17 to run. You are currently using Java 11.
      Your current JDK is located in C:\Program Files\Microsoft\OpenJDK
      You can try some of the following options:
       - changing the IDE settings.
       - changing the JAVA_HOME environment variable.
       - changing `org.gradle.java.home` in `gradle.properties`.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 20s

java.io.IOException: Unable to run java: com.adobe.air.ADTException: gradle tool failed:
FAILURE: Build failed with an exception.

* Where:
Build file 'C:\Users\user\AppData\Local\Temp\136975d6-c005-45d1-8b89-c13d3a0aa1a1\app\build.gradle' line: 1

* What went wrong:
An exception occurred applying plugin request [id: 'com.android.application']
> Failed to apply plugin 'com.android.internal.application'.
   > Android Gradle plugin requires Java 17 to run. You are currently using Java 11.
      Your current JDK is located in C:\Program Files\Microsoft\OpenJDK
      You can try some of the following options:
       - changing the IDE settings.
       - changing the JAVA_HOME environment variable.
       - changing `org.gradle.java.home` in `gradle.properties`.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 20s
```

Thanks

**Comment by ghost** [comment, marked as answer]
I had the same issue. How I fix it:
- download JDK 17 (I am using Android Studio gradle settings window for this)
<img width="982" alt="Screenshot 2024-06-03 at 22 56 10" src="https://github.com/airsdk/Adobe-Runtime-Support/assets/96975636/7977fa59-0fa6-4889-8642-9ca82d1963be">
- then set path to JDK in the adt.cfg file
<img width="974" alt="Screenshot 2024-06-03 at 22 59 08" src="https://github.com/airsdk/Adobe-Runtime-Support/assets/96975636/ce9ce83d-12e5-429c-8a07-dd1743aceb8e">
- then try to build again

**Reply by ajwfrost** [reply]
Yes, agreed - this is basically because the Android Gradle Plug-in that's needed in order to support Android 34 and above, has the requirement to be run on Java 17. Really not sure why, given Gradle itself is quite happy running on Java 8, and the AGP can still be set to use/emit Java code using version 8.

Possibly we could capture this and spit out a better warning earlier, seeing as we should be able to check what the requirements are for everything. It just means we're having to add more knowledge of those external toolchains into ADT, which I had hoped to avoid, but I guess there's no real downside to it (other than having to keep on top of their changes..!)

thanks

**Reply by jeuxdaily** [reply]
hi
I don't have this folder

Build Tools-->Gradle

**Comment by ghost** [comment]
I think, next time you encounter the issue, it is better to post it in the "Issues" section, not in "Discussions"

**Comment by pol2095** [comment]
I changed the Windows Environment Variables Path to OpenJDK 17,
now I have just this warning :
`NOTE: targets later than Android 33 need Gradle Plugin versions of 8 or more. Adjusting to 8.1.1`

**Reply by ozonerik** [reply]
> same issue, how to solved it

**Reply by ajwfrost** [reply]
@ozonerik  if you mean, you also see a note saying that it's adjusting to use AGP version 8.1.1 .. there's nothing to resolve, this just tells you that it's using the later version because of your use of Android 33 APIs.

**Comment by jonathanlevySW** [comment]
I'm using jenv on macos to manage my java environments. I've tried editing adt.cfg to point to the active one (being 17.0.11), but am still getting this error. I am able to update gradle in the Android Studio project, but this adds another step for testing.

**Reply by ajwfrost** [reply]
By "this error" do you mean the build is failing with this line:
```
Android Gradle plugin requires Java 17 to run. You are currently using Java 11.
```

We have a few options that we're looking into here:
1) We're going to try looking for Java 17 specifically when we know this is needed, and if we find that (via Java_Home or via Android Studio install etc) then we can switch to it
2) If we can't find it, we have to make a decision whether to fail the build - or whether to not update to the 8.x Android Gradle Plug-in. Which means it should/may build okay, but internally we see a warning that the plug-in has not been tested against this version of Android (which is what made us add this 'note' and do the update already..)

Maybe it's better to downgrade that to a warning, which I guess is how Android Studio currently treats it. We just assumed this would be setting us up for a fall later on, when some incompatibilities are introduced within Android, so at some point we're going to hit build errors unless we can find the v17 JDK...

thanks

**Reply by jonathanlevySW** [reply]
Apologies, my error is a bit different:

     [exec] * What went wrong:
     [exec] An exception occurred applying plugin request [id: 'com.android.library']
     [exec] > Failed to apply plugin 'com.android.internal.version-check'.
     [exec]    > Minimum supported Gradle version is 8.0. Current version is 7.5.

**Reply by ajwfrost** [reply]
@jonathanlevySW the latest AIR SDK should update this - the file to look at is:
lib/android/bin/gradle-wrapper.properties
If that doesn't already have the line
```
distributionUrl=https\://services.gradle.org/distributions/gradle-8.0-bin.zip
```
then please edit it to this value .. or else you can add a section to your application descriptor file if you prefer, within the `<android>` section, such as `<gradleVersion>8.0</gradleVersion>`

We're planning to update everyone to Gradle 8.0 because even the older Android Gradle Plug-ins work with that, and Gradle 8 will run even on Java 1.8. The web of dependencies is complex, but this one actually seems quite a straightforward choice :-)

thanks

**Reply by jonathanlevySW** [reply]
@ajwfrost , the gradleVersion descriptor was out of sync on my end, which was causing the problem. Thanks.

**Comment by jeuxdaily** [comment]
hi
now it's work fine
![ok](https://github.com/airsdk/Adobe-Runtime-Support/assets/14032996/0ea7a510-0c82-4862-94f4-e1625240d1f9)

**Comment by HanSpyMB** [comment]
Good morning,

Could someone explain to me step by step how to force the sdk to target API 34.
I'm exporting a bundle from sanimate and I'm a little lost.
All my exports target version 33 and I don't understand how to change it. I'll give you a screenshot of my android studio.
I have the latest version of AIR SDK 51.0.1.4
